### DISCUSSION TITLE
Reasons for keeping runtime warnings internal

### DISCUSSION THREAD

**Discussion by maximkrouk** [pointfreeco/swift-composable-architecture#2661] [Q&A] [answered]
I think runtime warnings might be helpful in a bunch of situations, it's pretty easy to copy the implementation, but is there any particular reason of keeping them internal other than "it's not really related to the TCA and therefore kept internal to avoid unnecessary API complications"

I'm considering adding it to https://github.com/capturecontext/swift-foundation-extensions, but kinda concerned about the [`@_transparent` attribute](https://github.com/apple/swift/blob/main/docs/TransparentAttr.md#when-should-you-use-_transparent). Feels like first point of the second section is related to OS dynamic libs, rather than swift packages, but the feedback is appreciated anyway 🫠

@stephencelis @mbrandonw @tgrapperon

**Comment by mbrandonw** [comment, marked as answer]
Hi @maximkrouk, there's no real reason. We are even using the `runtimeWarn` function in a number of our projects. We do have plans to extract it out into a library soon.

**Reply by maximkrouk** [reply]
Cool, thanks I'll leave the extraction for you then c:
